pkg/metrics/pprof.go

```diff
@@ -0,0 +1,18 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// +build pprof_enabled
+
+package metrics
+
+import "net/http/pprof"
+
+func init() {
+	// Extra handlers based on https://github.com/golang/go/blob/master/src/net/http/pprof/pprof.go#L80-L86
+	metricsExtraHandlers["/debug/pprof/"] = pprof.Index
```

Comment:
a newbie question @HeavyWombat 

So say the deployment is running with the image that does not have pprof enabled ... the "non-debug" image,
then a user tries to run `go tool pprof` against it, which presumably will hit these http endpoints,
is it confirmed that these handlers will not work / fail in a graceful way on the server side here ... i.e. we won't cause any panics etc.  in the shipwright controller ?

I did skim (but not read word for word) both https://github.com/golang/go/blob/master/src/net/http/pprof/pprof.go and https://blog.golang.org/pprof and could not gleam an answer.

thanks

otherwise I'm good

Reply:
went ahead and added the approve label

Reply:
also scanned the PR, though I did not open up every resolved conversation

Reply:
Sorry, I missed it this morning. When I was testing it, if you have the image with no `pprof` handlers, trying to reach the endpoints returns `404` errors. I have not seen any stack traces in the logs, so I guess we are fine from that point.

Reply:
no worries / sounds good / thanks